launch/launch/utilities/signal_management.py

```diff
@@ -235,6 +235,7 @@ def handle(
         :return: previous handler if any, otherwise None
         """
         signum = signal.Signals(signum)
+        signal.signal(signum, signal.default_int_handler)
```

Comment:
@ciandonovan thanks for catching this! I dropped the ball here, my bad.

That said, I don't think this is the complete fix. We should only override default signal handlers (chaining non-default ones), and we should restore whatever was in place on `AsyncSafeSignalManager` exit.

Reply:
In this case I'm not overriding any default handlers, as SIGINT is already handled by `signal.default_int_handler`, and SIGTERM isn't handled at all, so I'm just setting it to be handled by the same function.

Ideally both would be handled by some global _noop function that just passes, letting `signal.set_wakeup_fd` pick it up and do its job. Not sure the best way to implement that though. There was a _noop function in `test_signal_management.py` previously, but the assert fails if a signal handler was installed for SIGTERM as it's pointed to a different handler address.

Reply:
> In this case I'm not overriding any default handlers, as SIGINT is already handled by `signal.default_int_handler`, and SIGTERM isn't handled at all, so I'm just setting it to be handled by the same function.

Right, but we don't really know which signal number we will be required to handle, and even we did (or assumed so or constrained to be such and such), there is also the possibility that some other library, used by `launch` or using `launch`, may have set their own as well.

Reply:
So what should we do here? Is it ok to just override any existing signal handler with when this is called? How does this interact with `signal.set_wakeup_fd`? What if someone sets their own signal handler after initializing launch, will our handling still function?

Reply:
The only existing signal handler at this point is the one the Python binary sets itself for SIGINT to generate the `KeyboardInterrupt` exception, which I don't change as I point all handlers towards that.
The only other possibility is that someone could have created a handler within their launchfile? Not sure if that's even possible though, and would be terrible practice anyway.
Signal handlers they install in their nodes are of course separate, as signals are a per-process property, and launch is but the parent of the node children.
In terms of `signal.set_wakeup_fd`, nothing extra needs to be done. Installing the signal handlers here simply prevents the kernel from immediately terminating the process as is the common default disposition, and allows Python to then forward the signal through the fd.